demos/CMakeLists.txt

```diff
@@ -4,17 +4,20 @@
 
 cmake_minimum_required(VERSION 3.10)
 
+# Multi config generators such as Visual Studio ignore CMAKE_BUILD_TYPE. Multi config generators are configured with
+# CMAKE_CONFIGURATION_TYPES, but limiting options in it completely removes such build options
+if(NOT GENERATOR_IS_MULTI_CONFIG AND NOT DEFINED CMAKE_BUILD_TYPE)
+    message(STATUS "CMAKE_BUILD_TYPE not defined, 'Release' will be used")
+    # Setting CMAKE_BUILD_TYPE as CACHE must go before project(). Otherwise project() sets its value and set() doesn't take an effect
+    set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel ...")
```

Comment:
> RelWithDebInfo MinSizeRel

Do all dependencies support these modes properly? Does it checked somewhere?

Reply:
I don't know. This message is what `CMAKE_BUILD_TYPE` usually has, I just put a copy of it to mimic it. Would it be better if I truncated the message to `Choose the type of build`?

Reply:
What was the purpose of this change, anyway? Wouldn't it be sufficient to just add `NOT GENERATOR_IS_MULTI_CONFIG` to the original `if` command?

Reply:
Setting `CMAKE_BUILD_TYPE` as CACHE puts the correct value into cache. Earlier version shadows original `CMAKE_BUILD_TYPE` cache var.

Reply:
> Earlier version shadows original CMAKE_BUILD_TYPE cache var.

Yeah, but... is it actually a problem?

Reply:
This resulted in having two wars. While cached value claimed the type is `""`, as well as cache viewers did, in fact Release was generated.

Reply:
Yeah, but again... is that actually a problem? We're effectively just "symlinking" the blank configuration to the release configuration.

Reply:
Blank configuration seems to be a separate valid value, decoupling these two should be cleaner.

Reply:
Yeah, in the blank configuration none of the `*_DEBUG` or `*_RELEASE` variables are used, resulting in a build with neither optimizations nor debug info. :-)

I'm not convinced that we need to support such a configuration, but I also don't care enough to argue about it, so keep it like that if you want.